framework/source/class/qx/ui/embed/Iframe.js

```diff
@@ -67,13 +68,57 @@ qx.Class.define("qx.ui.embed.Iframe",
 
     this.__blockerElement = this._createBlockerElement();
 
-    if ((qx.core.Environment.get("engine.name") == "gecko"))
+    if ( qx.core.Environment.get("ecmascript.mutationobserver") )
     {
-      this.addListenerOnce("appear", function(e)
+      this.addListenerOnce("appear", function ()
+      {
+        var element = this.getContentElement().getDomElement();
+
+        // Mutation record check callback
+        var isDOMNodeInserted = function (mutationRecord)
+        {
+          var i;
+          // 'our' iframe was either added...
+          if (mutationRecord.addedNodes)
+          {
+            for (i = mutationRecord.addedNodes.length; i>=0; --i) {
+              if (mutationRecord.addedNodes[i] == element) {
+                return true;
+              }
+            }
+          }
+          // ...or removed
+          if (mutationRecord.removedNodes)
+          {
+            for (i = mutationRecord.removedNodes.length; i>=0; --i) {
+              if (mutationRecord.removedNodes[i] == element) {
+                return true;
+              }
+            }
+          }
+          return false;
+        };
+
+        var observer = new MutationObserver(function (mutationRecords)
+        {
+          if (mutationRecords.some(isDOMNodeInserted)) {
+            this._syncSourceAfterDOMMove();
+          }
+        }.bind(this));
+
+        // Observe parent element
+        var parent = this.getLayoutParent().getContentElement().getDomElement();
+        observer.observe(parent, { childList: true });
+
+      }, this);
+    }
+    else // !qx.core.Environment.get("ecmascript.mutationobserver")
+    {
+      this.addListenerOnce("appear", function ()
```

Comment:
@schnepe2 before your PR the mutation event was attached to all gecko browsers. Now it is attached to every browser NOT having the mutation observer. 
Before the PR the listener was not attached to IE, Safari and Chrome which both report to be webkit.

I think adding this listener should be guarded with
```javascript
if ((qx.core.Environment.get("engine.name") == "gecko"))
```
Or am I wrong here?